cmd/operator-sdk/add/api.go

```diff
@@ -84,19 +144,164 @@ Example:
 	}
 	apiCmd.Flags().BoolVar(&skipGeneration, "skip-generation", false,
 		"Skip generation of deepcopy and OpenAPI code and OpenAPI CRD specs")
+	apiCmd.Flags().StringVar(&helmChartRef, "helm-chart", "",
+		"Initialize helm operator with existing helm chart (<URL>, <repo>/<name>, or local path)")
+	apiCmd.Flags().StringVar(&helmChartVersion, "helm-chart-version", "",
+		"Specific version of the helm chart (default is latest versio)")
+	apiCmd.Flags().StringVar(&helmChartRepo, "helm-chart-repo", "",
+		"Chart repository URL for the requested helm chart")
 
 	return apiCmd
 }
 
 func apiRun(cmd *cobra.Command, args []string) error {
+
 	projutil.MustInProjectRoot()
+	log.Infof("Generating api version %s for kind %s.", apiVersion, kind)
+
+	operatorType := projutil.GetOperatorType()
+
+	switch operatorType {
+	case projutil.OperatorTypeGo:
+		if err := doGoAPIScaffold(); err != nil {
+			return err
+		}
+	case projutil.OperatorTypeAnsible:
+		if err := doAnsibleAPIScaffold(); err != nil {
+			return err
+		}
+	case projutil.OperatorTypeHelm:
+		if err := doHelmAPIScaffold(); err != nil {
+			return err
+		}
+	}
+	log.Info("API generation complete.")
+	return nil
+}
+
+func doAnsibleAPIScaffold() error {
```

Comment:
Have we not the func which is called currently to scaffold the files when we run `operator-sdk new --type`? Could we not centralize the implementation?

Reply:
The reason for this separation being: Not all resources need to be scaffolded when adding additional APIs. That being said we are using pre-existing functions where ever possible.

Reply:
OK, but:
- When we run  `operator-sdk new ` -> it  will  create  the  project  + scaffold the files
- When we run  `operator-sdk new api` -> it  will just do the second same step to scaffold the files
> The process/files scaffold to add API are the same always. 

Then, my suggestion here is: 
- Encapsulated the implementation to scaffold the files and then, call its methods in the commands executions. 

Which means that: 
- If something changes in the API scaffolding process called by these commands we will edit/change just one place instead of many in the code source. 

So, Is it make sense? Are we doing that?

Reply:
- First thing, just to clear any confusion, the command being worked upon is `operator-sdk add api` , I see that you have mentioned in above comment as `operator-sdk new api`.
- The scaffoded files are not the same for initial creation of a project, and for `add api`. Please take a look at what is being scaffolded for Ansible/helm from the PR. 
https://github.com/operator-framework/operator-sdk/blob/2827bb09483666bf16723b2089984d4f8611f302/cmd/operator-sdk/add/api.go#L174

https://github.com/operator-framework/operator-sdk/blob/2827bb09483666bf16723b2089984d4f8611f302/cmd/operator-sdk/add/api.go#L241

That is the reason, this has been separated. Does that make sense?

Reply:
I agree with @camilamacedo86 that there's probably an improvement to be made here. For now, let's add a TODO comment that captures what we want to do for a follow-on.